jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py

```diff
@@ -165,6 +169,11 @@ def test_unit_translation(cubeviz_helper):
     # check if units translated
     assert y_display_unit == u.MJy / u.sr
 
+    # get_data(use_display_units=True) should return surface brightness-like units
+    # NOTE: should be updated to /pix**2 or /sr
+    assert cubeviz_helper.app._get_display_unit('spectral_y') == u.MJy / u.pix
+    assert cubeviz_helper.get_data('Spectrum (sum)', use_display_units=True).unit == u.MJy / u.pix
```

Comment:
@gibsongreen - the first line works (although is inconsistent with the check above which is expecting /sr), and the second line is then failing the assertion by returning in MJy/sr.  Is this something going wrong in `flux_conversion` or do these need to wait for non-per-sr support?

Reply:
I didn't change any of the extraction code to append u.pix before it gets added to the data collection, so the data item for 'Spectrum (sum)'  is in MJy (without a denominator). The test failure I see running it locally is a conversion error between MJy / pix and MJy (which we don't have supported yet with the flux_conversion). The per pixel support right now is only when data loaded in is flux units, and I didn't add full support for it, that goal of #3111  was predicated on support for u.sr (with any other denominator being added in the generalization, and switching denominators possibly in that PR or in a future one). 

The y_display_unit test on L166 I believe is actually correct and the data displayed is per sr. There is this assumption in #3111 that the we don't allow change of the angle_unit value. Below is the line where this assumption is made. This support was going to be in the generalization ticket, and since this is the case, when you translate, you still get .pix in this case which for the spectral_y test on L174.

https://github.com/gibsongreen/jdaviz/blob/96b92634efda8551cc83624cb2af7d36b0543fba/jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py#L177

Reply:
@cshanahan1 - we now updated this test to have the _input_ cube be in MJy/sr and then the rest of the logic falls into place.  Would your future PR (adding support for non-steradian angle units) be able to re-introduce a test-case with input in flux units, test that they are coerced to flux/pix**2, and then test any of this similar logic that you think needs extra coverage?